### ISSUE TITLE
Fix "Detected package downgrade" when building 3.1.103.2-SDK using N-1 source-built or bootstrap flow

### ISSUE THREAD

**Issue by dagood** [dotnet/source-build#1570]
See https://gitter.im/dotnet/source-build?at=5e9f28efaeb59a22add10ab3 and onwards for context.

In 3.1.103.2-SDK and presumably 3.1.102, Microsoft.CodeAnalysis.Common/3.4.1 is source-built with this dependency:

```
<dependency id="System.Runtime.CompilerServices.Unsafe" version="4.5.2" exclude="Build,Analyzers" />
```

However, in `Private.SourceBuilt.Artifacts.0.1.0-3.1.101.2.tar.gz`, Microsoft.CodeAnalysis.Common/3.4.1 has this dependency:

```
<dependency id="System.Runtime.CompilerServices.Unsafe" version="4.7.0-preview3.19551.4" exclude="Build,Analyzers" />
```

Not using the Private.SourceBuilt.Artifacts version results in this error when building Arcade:

```
Detected package downgrade: System.Runtime.CompilerServices.Unsafe from 4.7.0 to 4.5.2. Reference the package directly from the project to select a different version. 
 Microsoft.DotNet.CodeAnalysis -> Microsoft.CodeAnalysis.CSharp 3.4.1 -> Microsoft.CodeAnalysis.Common 3.4.1 -> System.Text.Encoding.CodePages 4.7.0 -> System.Runtime.CompilerServices.Unsafe (>= 4.7.0) 
 Microsoft.DotNet.CodeAnalysis -> Microsoft.CodeAnalysis.CSharp 3.4.1 -> Microsoft.CodeAnalysis.Common 3.4.1 -> System.Runtime.CompilerServices.Unsafe (>= 4.5.2)
```

We should upgrade `Private.SourceBuilt.Artifacts` so that we repro the error, then fix it.

If my understanding is correct, this blocks bootstrapping and N-1 flow from 3.1.102 => 3.1.103. (Tentative workaround: ignore the warning.)

/cc @crummel @dseefeld

**Comment by omajid** [comment]
> Workaround: ignore the warning.

Would that produce a source-built SDK that is functional?

**Comment by dagood** [comment]
I'm not quite sure... I don't think this will affect the version of the compiler (or its deps) that we ship, since this is an Arcade package and we rebuild CoreFX + Roslyn later on, but I don't know if there are any more subtle effects to look out for.

I also don't know yet if more errors might show up later in the build, maybe without a workaround.

I'm pinging the team about this, seems like a critical issue to me so far.

**Comment by dseefeld** [comment]
@crummel to update previously source built packages and add this as a step to the release checklist.

**Comment by MichaelSimons** [comment]
[Triage] This issue has been fixed for 3.1.
